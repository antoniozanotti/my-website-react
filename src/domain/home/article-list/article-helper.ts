import { NodeTypeEnum } from "@/enums/node-type-enum";
import { ArticleType } from "@/types/article-type";
import { Link } from "@/types/link";

export function getParagraphs(article: ArticleType): string[] {
  let paragraphs: string[] = [];
  article.achievements.json.content.forEach((p) => {
    if (
      p.content === undefined ||
      p.content[0].value === "" ||
      p.content[0].value === undefined
    ) {
      return;
    }
    paragraphs.push(p.content[0].value);
  });
  return paragraphs;
}

export function getLinks(article: ArticleType): Link[] {
  let links: Link[] = [];
  article.links?.json.content
    .filter((link) => link.nodeType == NodeTypeEnum["unordered-list"])[0]
    .content?.forEach((link) => {
      if (
        link.content === undefined ||
        link.content[0].content === undefined
      ) {
        return;
      }
      let linkContent = link.content[0].content.filter(
        (l) => l.nodeType == NodeTypeEnum["hyperlink"]
      )[0];
      if (linkContent.content === undefined) {
        return;
      }
      links.push({
        link: linkContent.data.uri ?? "",
        label: linkContent.content[0].value ?? "",
      });
    });
  return links;
}

export function getSkills(article: ArticleType): string[] {
  let skills: string[] = [];
  article.skillsCollection?.items.map((item) => {
    skills.push(item.title);
  });
  return skills;
}