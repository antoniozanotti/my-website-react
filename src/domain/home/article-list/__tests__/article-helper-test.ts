import { getLinks, getParagraphs, getSkills } from "../article-helper";
import { ExperienceMocks } from "../../experience/__tests__/experience-mocks";

describe("ArticleHelper", () => {
  it("should return paragraphs", async () => {
    const paragraphs = getParagraphs(ExperienceMocks.positions[0]);

    expect(paragraphs).toMatchSnapshot();
  });

  it("should return links", async () => {
    const links = getLinks(ExperienceMocks.positions[0]);

    expect(links).toMatchSnapshot();
  });

  it("should return skills", async () => {
    const skills = getSkills(ExperienceMocks.positions[0]);

    expect(skills).toMatchSnapshot();
  });
});
