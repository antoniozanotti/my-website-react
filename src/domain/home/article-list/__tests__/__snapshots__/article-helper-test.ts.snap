// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHelper should return links 1`] = `
[
  {
    "label": "EcoSmart Fire (United States)",
    "link": "https://ecosmartfire.com/",
  },
]
`;

exports[`ArticleHelper should return paragraphs 1`] = `
[
  "In 2023, I designed the transition to the JAMstack architecture, which will gradually replace the Magento platform by APIs. This migration will result in significant gains in application performance, and will provide greater independence for content managers. For this, I created a system using Vue, Vite, Pinia, Typescript, UNOCSS+Tailwind CSS, Storybook, urql for team collaborative development. I proposed improvements to data modeling in Hygraph (GraphQL Headless CMS), to improve performance and maintain the platform's flexibility.",
]
`;

exports[`ArticleHelper should return skills 1`] = `
[
  "Front-End Development",
  "Software Architecture",
  "React",
  "Next.js",
  "Node.js",
  "TypeScript",
  "JavaScript",
  "TanStack Query",
  "Axios",
  "Test Automation",
  "Playwright",
  "Regex",
  "Data Modeling",
  "PostgreSQL",
  "Prisma",
  "SQL",
  "HTML",
  "Tailwind CSS",
  "CSS",
  "NextUI",
  "SEO",
  "Software Development",
  "Documentation",
  "Faker.js",
  "Git",
  "npm",
]
`;
